### ISSUE TITLE
Add deduplication

### ISSUE THREAD

**Issue by dsmilkov** [lilacai/lilac#801]
We should return a new copy of a dataset, or at the minimum enrich the dataset with a "dedup_drop" column for the rows that need to be dropped.

We should support identical, min-hash and semantic dedup. The dedup should take a list of fields (e.g. dedup "question" and "answer" as a combination) and it should dedup when each of the individual fields is a near-duplicate.

**Comment by nsthorat** [comment]
I don't think we should make a new dataset, rather we should just tag for deletion. It gives the user the ability to manually inspect and undo (and fits with the same mental model of statefulness)
